src/main/java/org/jabref/logic/util/io/FileNameUniqueness.java

```diff
@@ -41,4 +48,74 @@ public static String getNonOverWritingFileName(Path targetDirectory, String file
+     *         false when there is no "similar" file name or the content is different from that of files with "similar" name
+     * @throws IOException Fail when the file is not exist or something wrong when reading the file
+     */
+    public static boolean isDuplicatedFile(Path directory, Path fileName, DialogService dialogService) throws IOException {
+
+        Objects.requireNonNull(directory);
+        Objects.requireNonNull(fileName);
+        Objects.requireNonNull(dialogService);
+
+        String extensionSuffix = FileUtil.getFileExtension(fileName).orElse("");
+        extensionSuffix = extensionSuffix.equals("") ? extensionSuffix : "." + extensionSuffix;
+        String newFilename = FileUtil.getBaseName(fileName) + extensionSuffix;
+
+        String fileNameWithoutDuplicated = eraseDuplicateMarks(FileUtil.getBaseName(fileName));
+        String originalFileName = fileNameWithoutDuplicated + extensionSuffix;
+
+        if (newFilename.equals(originalFileName)) {
+            return false;
+        }
+
+        Path originalFile = directory.resolve(originalFileName);
+        Path duplicateFile = directory.resolve(fileName);
+        int counter = 1;
+
+        while (Files.exists(originalFile)) {
+            if (com.google.common.io.Files.equal(originalFile.toFile(), duplicateFile.toFile())) {
+                if (duplicateFile.toFile().delete()) {
+                    dialogService.notify(Localization.lang("File '%1' is a duplicate of '%0'. Keeping '%0'", originalFileName, fileName));
+                } else {
+                    dialogService.notify(Localization.lang("File '%1' is a duplicate of '%0'. Keeping both due to deletion error", originalFileName, fileName));
+                }
+                return true;
+            }
+
+            originalFileName = fileNameWithoutDuplicated +
+                    " (" + counter + ")"
+                    + extensionSuffix;
+            counter++;
+
+            if (newFilename.equals(originalFileName)) {
+                return false;
+            }
+            originalFile = directory.resolve(originalFileName);
+        }
+        return false;
+    }
+
+    /**
+     * This is the opposite function of getNonOverWritingFileName
+     * It will recover the file name to origin if it has duplicate mark such as " (1)"
+     * change the String whose format is "xxxxxx (number)" into "xxxxxx", while return the same String when it does not match the format
+     * This is the opposite function of getNonOverWritingFileName
+     *
+     * @param fileName Suggested name for the file without extensionSuffix, if it has duplicate file name with other file, it will end with something like " (1)"
+     * @return Suggested name for the file without extensionSuffix and duplicate marks such as " (1)"
+     */
+    public static String eraseDuplicateMarks(String fileName) {
+        Pattern p = Pattern.compile("(.*) \\(\\d+\\)");
```

Comment:
move this Pattern.compile to a static field in the class e.g. private static final Pattern DUPLICATE_MARK_PATTERN = Pattern.compile . The rest is good so far